src/scipp/core/binning.py

```diff
@@ -95,6 +98,13 @@ def make_binned(x: _cpp.DataArray,
         groups = []
     if edges is None:
         edges = []
+    if isinstance(x, Variable):
+        coords = edges + groups
```

Comment:
```suggestion
        coords = [*edges , *groups]
```
to make it work with arbitrary sequences as the type hints claim.